source/renderer/app/components/widgets/forms/PinCode.js

```diff
@@ -31,42 +39,258 @@ export default class PinCode extends Component<Props> {
   inputsRef = [];
   focusKey = 0;
   add = false;
+  fromBackspace = false;
+
+  state = {
+    isBackSpace: false,
+    focusKeyChanged: false,
+    focusIsUpdated: false,
+  };
+
+  hasInputNewValue = (inputNewValue: string, key: number) => {
+    const { value } = this.props;
+    const valueIsEmpty =
+      !value[key] || value[key] === '' || inputNewValue === '';
+    const valueHasChanged =
+      inputNewValue &&
+      inputNewValue.length === 1 &&
+      value[key] !== inputNewValue;
+    return valueIsEmpty || valueHasChanged;
+  };
+
+  isInputValueSelected = (input: Field) => {
+    const { inputElement } = input;
+    const { current } = inputElement;
+    return current.selectionStart !== current.selectionEnd;
+  };
+
+  isInputFieldDisabled = (
+    index: number,
+    value: Array<string>,
+    disabled: boolean
+  ) => {
+    return (
+      disabled ||
+      (index !== 0 &&
+        (!value ||
+          (!value.length && !value[index - 1] && value[index] && !!value[this.focusKey]) ||
+          (value.length &&
+            index > this.focusKey &&
+            !value[index] &&
+            !value[this.focusKey])))
+    );
+  };
 
-  onChange = (inputValue: ?number, key: number) => {
+  onChange = (inputValue: ?number | ?string, key: number) => {
     const { value, onChange } = this.props;
+    const { isBackSpace } = this.state;
     const inputNewValue =
-      inputValue !== null && inputValue !== undefined
+      inputValue !== null && inputValue !== undefined && !isNaN(inputValue)
```

Comment:
you can convert this `inputValue !== null && inputValue !== undefined` to just `inputValue && !isNaN(inputValue)`

Reply:
Done